[CI] MultiClustersIT testLookupJoinAliases failing

**Build Scans:**
- [elasticsearch-periodic #9335 / openjdk23_8.19.0_java-matrix-bwc](https://gradle-enterprise.elastic.co/s/mfsq3rr322pok)
- [elasticsearch-pull-request #80247 / 9.0.4_part_3_bwc-snapshots](https://gradle-enterprise.elastic.co/s/nrq7m6xnam4fs)
- [elasticsearch-pull-request #80247 / 9.2.0_part_3_bwc-snapshots](https://gradle-enterprise.elastic.co/s/pg34hwvf5sfuc)
- [elasticsearch-pull-request #80247 / 8.19.0_part_3_bwc-snapshots](https://gradle-enterprise.elastic.co/s/wbjyq6uw2hw5g)
- [elasticsearch-pull-request #80197 / 9.1.0_part_3_bwc-snapshots](https://gradle-enterprise.elastic.co/s/dnyhq5mnbmv4c)
- [elasticsearch-pull-request #80194 / 9.0.4_part_3_bwc-snapshots](https://gradle-enterprise.elastic.co/s/yltgs2jksuouk)
- [elasticsearch-pull-request #80194 / 8.19.0_part_3_bwc-snapshots](https://gradle-enterprise.elastic.co/s/s2gge7e7kuz3a)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:multi-clusters:v8.19.0#bwcTest" -Dtests.class="org.elasticsearch.xpack.esql.ccq.MultiClustersIT" -Dtests.method="testLookupJoinAliases" -Dtests.seed=DB58AB280F4C0499 -Dtests.bwc=true -Dtests.locale=en-FJ -Dtests.timezone=GB-Eire -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.ccq.MultiClustersIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testLookupJoinAliases'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Expected a map containing
documents_found: <638>
  values_loaded: <638>
           took: <208>
     is_partial: <false>
      _clusters: <{running=0, total=2, details={remote_cluster={_shards={total=1, failed=0, successful=1, skipped=0}, took=192, indices=test-remote-index, status=successful}, (local)={_shards={total=5, failed=0, successful=5, skipped=0}, took=207, indices=test-local-index, status=successful}}, failed=0, partial=0, successful=2, skipped=0}>
        columns: a list containing
                0: a map containing
               name: "c"
               type: "long"
         values: a list containing
                0: a list containing
                  0: expected <638> but was <640>
        profile: <{planning={took_millis=45, start_millis=1752415687062, stop_millis=1752415687107, took_nanos=45108682}, plans=[{cluster_name=remote_cluster, node_name=remote_cluster-1, description=data, plan=ExchangeSinkExec[[$$c$count{r}#27, $$c$seen{r}#28],true]
[truncated]
```

**Issue Reasons:**
- [main] 7 failures in test testLookupJoinAliases (0.9% fail rate in 783 executions)
- [main] 2 failures in step 8.19.0_part_3_bwc-snapshots (2.7% fail rate in 75 executions)
- [main] 2 failures in step 9.0.4_part_3_bwc-snapshots (2.7% fail rate in 75 executions)
- [main] 3 failures in pipeline elasticsearch-pull-request (3.7% fail rate in 81 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -354,9 +354,6 @@ tests:
 - class: org.elasticsearch.search.SearchWithRejectionsIT
   method: testOpenContextsAfterRejections
   issue: https://github.com/elastic/elasticsearch/issues/130821
-- class: org.elasticsearch.xpack.esql.ccq.MultiClustersIT
-  method: testLookupJoinAliases
-  issue: https://github.com/elastic/elasticsearch/issues/131166
 - class: org.elasticsearch.packaging.test.DockerTests
   method: test090SecurityCliPackaging
   issue: https://github.com/elastic/elasticsearch/issues/131107
@@ -408,9 +405,6 @@ tests:
 - class: org.elasticsearch.compute.lucene.read.SortedSetOrdinalsBuilderTests
   method: testReader
   issue: https://github.com/elastic/elasticsearch/issues/131573
-- class: org.elasticsearch.xpack.esql.ccq.MultiClustersIT
-  method: testLookupJoinAliasesSkipOld
-  issue: https://github.com/elastic/elasticsearch/issues/131697
 - class: org.elasticsearch.packaging.test.DockerTests
   method: test151MachineDependentHeapWithSizeOverride
   issue: https://github.com/elastic/elasticsearch/issues/123437

--- a/x-pack/plugin/esql/qa/server/multi-clusters/src/javaRestTest/java/org/elasticsearch/xpack/esql/ccq/MultiClustersIT.java
+++ b/x-pack/plugin/esql/qa/server/multi-clusters/src/javaRestTest/java/org/elasticsearch/xpack/esql/ccq/MultiClustersIT.java
@@ -128,14 +128,12 @@ public class MultiClustersIT extends ESRestTestCase {
              }
             """;
         var randomDocsData = new ArrayList<Integer>();
-        var lookupDocs = IntStream.range(0, between(1, 5))
-            .mapToObj(n -> {
-                String color = randomFrom("red", "yellow", "green");
-                int data = randomValueOtherThanMany(i -> randomDocsData.contains(i), () -> randomIntBetween(1, 1000));
-                randomDocsData.add(data);
-                return new Doc(n, color, data);
-            })
-            .toList();
+        var lookupDocs = IntStream.range(0, between(1, 5)).mapToObj(n -> {
+            String color = randomFrom("red", "yellow", "green");
+            int data = randomValueOtherThanMany(i -> randomDocsData.contains(i), () -> randomIntBetween(1, 1000));
+            randomDocsData.add(data);
+            return new Doc(n, color, data);
+        }).toList();
         createIndex(
             localClient,
             lookupIndexLocal,

--- a/x-pack/plugin/esql/qa/server/multi-clusters/src/javaRestTest/java/org/elasticsearch/xpack/esql/ccq/MultiClustersIT.java
+++ b/x-pack/plugin/esql/qa/server/multi-clusters/src/javaRestTest/java/org/elasticsearch/xpack/esql/ccq/MultiClustersIT.java
@@ -127,8 +127,14 @@ public class MultiClustersIT extends ESRestTestCase {
                "morecolor": { "type": "keyword" }
              }
             """;
+        var randomDocsData = new ArrayList<Integer>();
         var lookupDocs = IntStream.range(0, between(1, 5))
-            .mapToObj(n -> new Doc(n, randomFrom("red", "yellow", "green"), randomIntBetween(1, 1000)))
+            .mapToObj(n -> {
+                String color = randomFrom("red", "yellow", "green");
+                int data = randomValueOtherThanMany(i -> randomDocsData.contains(i), () -> randomIntBetween(1, 1000));
+                randomDocsData.add(data);
+                return new Doc(n, color, data);
+            })
             .toList();
         createIndex(
             localClient,

